cpp/memilio/math/integrator.cpp

```diff
@@ -36,26 +36,27 @@ Eigen::Ref<Eigen::VectorXd> OdeIntegrator::advance(const DerivFunction& f, const
 
     bool step_okay = true;
 
-    double t = t0;
-    size_t i = results.get_num_time_points() - 1;
+    double dt_restore = 0; // used to restore dt if dt was decreased to reach tmax
```

Comment:
Looks good. Also checked for some ODE examples and behavior stays the same. 
One comment. We calculate `const size_t nb_steps = (int)(ceil((tmax - t0) / dt));`

ceil returns a double. We cast double -> int -> size_t. Why not directly double -> size_t?

Reply:
Thank you!
While the two casts should not matter - at the sizes it would, we just might run out of memory - the int cast is unnecessary and can be removed.